geometry/src/main/java/coppercore/Line.java

```diff
@@ -0,0 +1,31 @@
+package coppercore.geometry;
+
+public class Line {
+    private double x1;
+    private double y1;
+    private double x2;
+    private double y2;
+    private double slope;
+
+    public Line(double x1, double y1, double x2, double y2) {
+        this.x1 = x1;
+        this.y1 = y1;
+        this.x2 = x2;
+        this.y2 = y2;
+        this.slope = (y2 - y1) / (x2 - x1);
+    }
+
+    public Line(Point point1, Point point2) {
+        this.x1 = point1.getX();
+        this.y1 = point1.getY();
+        this.x2 = point2.getX();
+        this.y2 = point2.getY();
+        this.slope = (this.y2 - this.y1) / (this.x2 - this.x1);
+    }
+
+    public boolean isPointOnLine(Point point) {
+        return point.getY() == point.getX() * slope
+                && (x1 < point.getX() && point.getX() < x2)
+                && (y1 < point.getY() && point.getY() < y2);
+    }
+}
```

Comment:
Say we have a line that goes from (0,1) to (2,3)
The slope of that line would be 1.
However, with this function, if we test whether the point (0.5, 1.5) is on the line, the condition:
`point.getY() == point.getX() * slope` will be false because: 1.5 != 0.5 * 1, even though the point is on the line.
Could we reimplement this function to work differently? What @jkleiber wrote above should work.